### PR TITLE
Add unit tests

### PR DESCRIPTION
**Pull request by V1talie**

### CODE REVIEW

File: src/test/java/com/endava/internship/collections/StudentListTests.java [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+                () -> assertThat(studentList.iterator().next()).isEqualTo(student0),
+                () -> assertThat(studentList.get(0)).isEqualTo(student0),
+                () -> assertThat(studentList.toArray(new Student[1])).containsOnly(student0),
+                () -> assertThat(studentList).containsExactlyElementsOf(stList),
+                () -> assertThat(studentList.add(student1)).isEqualTo(stList.add(student1))
+        );
+    }
+
+    @Test
+    public void removingAnElement_shouldRemoveItFromList() {
```
Comment (GAlexandru-be) [inline src/test/java/com/endava/internship/collections/StudentListTests.java]: remove public modifier

**Comment by GAlexandru-be** [comment]
Use SonarLint and fix all warnings

File: src/test/java/com/endava/internship/collections/StudentListTests.java [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+    @BeforeEach
+    void setUp() {
+        studentList = new StudentList<>();
+        stList = new ArrayList<>();
+    }
+
+    @Test
+    void initialSize_shouldBeZero() {
+        final int size = studentList.size();
+        assertEquals(0, size);
```
Comment (GAlexandru-be) [inline src/test/java/com/endava/internship/collections/StudentListTests.java]: assertEquals(0, studentList.size());

File: src/test/java/com/endava/internship/collections/StudentListTests.java (hunk 1 above)
Comment (V1talie) [reply src/test/java/com/endava/internship/collections/StudentListTests.java]: Done

File: src/test/java/com/endava/internship/collections/StudentListTests.java [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+
+        ListIterator<Student> listIterator = studentList.listIterator();
+        listIterator.nextIndex();
+        assertThat(listIterator.previous()).isEqualTo(student0);
+
+    }
+
+    @ParameterizedTest
+    @MethodSource("studentProvider")
+    void setChangeTest(List<Student> student) {
```
Comment (GAlexandru-be) [inline src/test/java/com/endava/internship/collections/StudentListTests.java]: students*

File: src/test/java/com/endava/internship/collections/StudentListTests.java [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+    }
+
+    @Test
+    public void ifAnElementDoesNotExistIndexOf_shouldReturnNegativeOne() {
+        studentList.add(student4);
+        studentList.add(student3);
+        assertThat(studentList.indexOf(student1)).isEqualTo(-1);
+    }
+
+    static Stream<Arguments> studentProvider() {
```
Comment (GAlexandru-be) [inline src/test/java/com/endava/internship/collections/StudentListTests.java]: private ..

File: src/test/java/com/endava/internship/collections/StudentListTests.java [hunk 5]
```diff
@@ -0,0 +1,367 @@
...
+    }
+
+    @Test
+    void addAll_shouldAddAllElementsFromSourceListToDestinationList() {
+        StudentList<Student> studentListSource = new StudentList<>();
+        StudentList<Student> studentListDest = new StudentList<>();
+        List<Student> arrayListSource = new ArrayList<>();
+        List<Student> arrayListDest = new ArrayList<>();
+
+        Student student0 = new Student("student", LocalDate.parse("2000-10-05"), "none");
```
Comment (GAlexandru-be) [inline src/test/java/com/endava/internship/collections/StudentListTests.java]: why are you creating new students here if you have some students above?

File: src/test/java/com/endava/internship/collections/StudentListTests.java [hunk 6]
```diff
@@ -0,0 +1,367 @@
...
+    }
+
+    @Test
+    public void iteratorPrevious_shouldReturnElementAtPreviousPosition(){
+        studentList.add(student0);
+        studentList.add(student1);
+        studentList.add(student2);
+
+        ListIterator<Student> listIterator = studentList.listIterator();
+        listIterator.nextIndex();
```
Comment (GAlexandru-be) [inline src/test/java/com/endava/internship/collections/StudentListTests.java]: your nextIndex is not working correctly , you should return the index,
Try to do the same test for ArrayList from collections and see if they are working the same

File: src/test/java/com/endava/internship/collections/StudentListTests.java [hunk 7]
```diff
@@ -0,0 +1,368 @@
...
+
+    @Test
+     void ifAnElementDoesNotExistIndexOf_shouldReturnNegativeOne() {
+        studentList.add(student4);
+        studentList.add(student3);
+        assertThat(studentList.indexOf(student1)).isEqualTo(-1);
+    }
+
+    static Stream<Arguments> studentProvider() {
+        Student student0 = new Student("student", LocalDate.parse("2000-10-05"), "none");
```
Comment (GAlexandru-be) [inline src/test/java/com/endava/internship/collections/StudentListTests.java]: remove these also

File: src/test/java/com/endava/internship/collections/StudentListTests.java (hunk 2 above)
Comment (V1talie) [reply src/test/java/com/endava/internship/collections/StudentListTests.java]: Done

File: src/test/java/com/endava/internship/collections/StudentListTests.java (hunk 3 above)
Comment (V1talie) [reply src/test/java/com/endava/internship/collections/StudentListTests.java]: Done

File: src/test/java/com/endava/internship/collections/StudentListTests.java (hunk 4 above)
Comment (V1talie) [reply src/test/java/com/endava/internship/collections/StudentListTests.java]: Done

File: src/test/java/com/endava/internship/collections/StudentListTests.java (hunk 5 above)
Comment (V1talie) [reply src/test/java/com/endava/internship/collections/StudentListTests.java]: Done(my bad)

File: src/test/java/com/endava/internship/collections/StudentListTests.java (hunk 6 above)
Comment (V1talie) [reply src/test/java/com/endava/internship/collections/StudentListTests.java]: Done(fixed)

File: src/test/java/com/endava/internship/collections/StudentListTests.java (hunk 7 above)
Comment (V1talie) [reply src/test/java/com/endava/internship/collections/StudentListTests.java]: Done

**Comment by V1talie** [comment]
image

**Review by GAlexandru-be — APPROVED** [review]

**Pull request merged**